plugins/modules/notification/discord.py

```diff
@@ -0,0 +1,172 @@
+    description:
+      - The webhook id
+      - Format from discord webhook URL: C(/webhooks/{webhook.id}/{webhook.token})
+    required: yes
+    type: str
+  webhook_token:
+    description:
+      - The webhook token
+      - Format from discord webhook URL: C(/webhooks/{webhook.id}/{webhook.token})
+    required: yes
+    type: str
+  content:
+    description:
+      - The the message contents
+    type: str
+  username:
+    description:
+      - Override the default username of the webhook
+    type: str
+  avatar_url:
+    description:
+      - Override the default avatar of the webhook
+    type: str
+  tts:
+    description:
+      - True if this is a TTS message
+      - Text to speech message
+    type: bool
+  embeds:
+    description:
+      - Embedded rich content
+      - Send messages as Embeds
+    type: str
+'''
+
+EXAMPLES = """
+- name: Send a message to the discord channel
+  community.general.discord:
+    webhook_id: "00000"
+    webhook_token: "XXXYYY"
+    content: "This is a message from ansible"
+
+- name: Send a message to the discord channel with specific username and avatar
+  community.general.discord:
+    webhook_id: "00000"
+    webhook_token: "XXXYYY"
+    content: "This is a message from ansible"
+    username: Ansible
+    avatar_url: "https://docs.ansible.com/ansible/latest/_static/images/logo_invert.png"
+
+- name: Send a embedded message to the discord channel
+  community.general.discord:
+    webhook_id: "00000"
+    webhook_token: "XXXYYY"
+      embeds:
+        - title: "Embedded message"
+          description: "This is an embedded message"
+          footer:
+            text: "Author: Ansible"
+          image:
+            url: "https://docs.ansible.com/ansible/latest/_static/images/logo_invert.png"
+"""
+
+RETURN = """
+http_code:
+  description:
+    - Response Code returned by discord API.
+  returned: always
+  type: int
+  sample: 204
+"""
+
+
+def discord_text_msg(module):
+
+    webhook_id = module.params['webhook_id']
+    webhook_token = module.params['webhook_token']
+    content = module.params['content']
+    user = module.params['username']
+    avatar_url = module.params['avatar_url']
+    tts = module.params['tts']
+    embeds = module.params['embeds']
+
+    headers = {
+        'content-type': 'application/json'
+    }
+
+    url = "https://discord.com/api/webhooks/%s/%s" % (
+        webhook_id, webhook_token)
+
+    payload = {
+        'content': content,
+        'username': user,
+        'avatar_url': avatar_url,
+        'tts': tts,
+        'embeds': embeds,
+    }
+
+    payload = module.jsonify(payload)
+
+    response, info = fetch_url(module, url, data=payload, headers=headers)
+    return response, info
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            webhook_id=dict(type='str', required=True),
+            webhook_token=dict(type='str', required=True, no_log=True),
+            content=dict(type='str', required=False),
```

Comment:
It's unnecessary to state default value of `required`.
```suggestion
            content=dict(type='str'),
```

Reply:
Removed all `required=False`